vespalib/src/vespa/vespalib/net/emulated_epoll.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "wakeup_pipe.h"
+#include <poll.h>
+#include <map>
+#include <mutex>
+
+#define EPOLLERR POLLERR
+#define EPOLLHUP POLLHUP
+#define EPOLLIN POLLIN
+#define EPOLLOUT POLLOUT
+
+namespace vespalib {
+
+// structure describing which event occurred.
+struct epoll_event
+{
+    struct {
+        void *ptr;
+    } data;
+    uint32_t events;
+};
+
+/**
+ * The Epoll class is a thin wrapper around basic emulation of the epoll
+ * related system calls.
+ **/
+class Epoll
+{
+private:
+    std::mutex _monitorlock;
```

Comment:
the use of 'monitor' here is slightly confusing, consider using 'monitored_lock' instead